src/SFML/Audio/SoundFileReaderMp3.cpp

```diff
@@ -0,0 +1,141 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2021 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#define MINIMP3_IMPLEMENTATION // Minimp3 control define, places implementation in this file.
+#define NOMINMAX          // To avoid windows.h and std::min issue
+#define MINIMP3_NO_STDIO  // Minimp3 control define, eliminate file manipulation code which is useless here
+#include <minimp3_ex.h>
+#undef NOMINMAX
+#undef MINIMP3_NO_STDIO
```

Comment:
The problem here is that `MINIMP3_IMPLEMENTATION` is still defined when including `minimp3_ex.h` a second time in  `SoundFileReaderMp3.hpp` just below. There is no include guard for the implementation part, so the implementation is repeated.
I can see a few solutions:
- `#undef MINIMP3_IMPLEMENTATION`
- Move this include for the implementation after having included `SoundFileReaderMp3.hpp`.
- Add an include guard in `minimp3_ex.h` as it was already done in `minimp3.h` (see: https://github.com/lieff/minimp3/commit/644e0fb7fed34f803b6634f72e5ad8cc20a520f7).

Reply:
@lieff could we add an include guard in `minimp3_ex.h` as well?

Reply:
Sure, added upstream https://github.com/lieff/minimp3/commit/50d2aaf360a53653b718fead8e258d654c3a7e41

Reply:
I've updated the header, but seems like that's not enough, the unity build still keeps failing... 🤔

Reply:
Here is what happens in a unity build batch:
```c++
#include <minimp3_ex.h>
```
This pulls `minimp3_ex.h` declarations.
This declarations part includes `minimp3.h` which pulls `minimp3.h` declarations.

Then, later:
```c++
#define MINIMP3_IMPLEMENTATION
#include <minimp3_ex.h>
```

As they were already included, `minimp3_ex.h` declarations are skipped.
It also skips `#includes <minimp3.h>`, so `minimp3.h` implementation is not included.
Then comes `minimp3_ex.h` implementation which cannot find `minimp3.h` implementation, as it was not included.

A workaround is to include both `minimp3.h` and `minimp3_ex.h`, but it can be fixed in minimp3 directly with the following patch. (I also found that size_t type was missing.)  @lieff 
```patch
diff --git a/extlibs/headers/minimp3/minimp3_ex.h b/extlibs/headers/minimp3/minimp3_ex.h
index 4ccc5963..64d2f39f 100644
--- a/extlibs/headers/minimp3/minimp3_ex.h
+++ b/extlibs/headers/minimp3/minimp3_ex.h
@@ -7,6 +7,7 @@
     See <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
 #include "minimp3.h"
+#include <stddef.h> // size_t
 
 /* flags for mp3dec_ex_open_* functions */
 #define MP3D_SEEK_TO_BYTE   0      /* mp3dec_ex_seek seeks to byte in stream */
@@ -130,6 +131,8 @@ int mp3dec_ex_open_w(mp3dec_ex_t *dec, const wchar_t *file_name, int flags);
 
 #if defined(MINIMP3_IMPLEMENTATION) && !defined(_MINIMP3_EX_IMPLEMENTATION_GUARD)
 #define _MINIMP3_EX_IMPLEMENTATION_GUARD
+
+#include "minimp3.h"
 #include <limits.h>
 
 static void mp3dec_skip_id3v1(const uint8_t *buf, size_t *pbuf_size)
```

Reply:
vittorioromeo managed to solve the warnings